File: 2022/r/day04.r
# --- Day 4: Camp Cleanup ---
x <- read.table(text = gsub("-|,", " ", readLines("2022/data/input04.txt")))

# Part 1
sum((x[1] >= x[3] & x[2] <= x[4]) | (x[1] <= x[3] & x[2] >= x[4]))

# Part 2
sum((x[2] >= x[3] & x[1] <= x[4]))

File: 2022/julia/day04.jl
# --- Day 4: Camp Cleanup ---

parse_input = function(input_path)
    input = split.(readlines(input_path), (['-',','],))
    input_int = [parse.(Int, x) for x in input]
    return [[col[i] for col in input_int] for i in 1:4]
end

part01 = function(input_path)
    a, b, c, d  = parse_input(input_path)
    sum((a .>= c) .& (b .<= d) .| (a .<= c) .& (b .>= d))
end

part02 = function(input_path)
    a, b, c, d  = parse_input(input_path)
    sum((b .>= c) .& (a .<= d))
end

println("Answer 1: ", part01("2022/data/input04.txt"))
println("Answer 2: ", part02("2022/data/input04.txt"))
